src/v/kafka/server/handlers/incremental_alter_configs.cc

```diff
@@ -93,14 +93,14 @@ static void parse_and_set_shadow_indexing_mode(
         simode.op = cluster::incremental_update_operation::remove;
         break;
     case config_resource_operation::set:
-        simode.value
-          = string_switch<std::optional<model::shadow_indexing_mode>>(*value)
-              .match("no", std::nullopt)
-              .match("false", std::nullopt)
-              .match("yes", enabled_value)
-              .match("true", enabled_value)
-              .default_match(std::nullopt);
-        simode.op = cluster::incremental_update_operation::set;
+        simode.value = enabled_value;
+        simode.op
+          = string_switch<cluster::incremental_update_operation>(*value)
+              .match("no", cluster::incremental_update_operation::remove)
+              .match("false", cluster::incremental_update_operation::remove)
```

Comment:
i think that we should not remove property when caller wants to set a value, this way when default value is set to `true` and someone requests to set it to `false` it will never happen.

Reply:
fixed